data table tutorial (#304)
reflex-dev/reflex-web: docs/datatable-tutorial/simple_table.md

```python exec
import reflex as rx
from pcweb.templates.docpage import docdemobox, doclink
from pcweb.base_state import State
from pcweb.pages.docs.datatable_tutorial.states import DataTableState, DataTableState2
```

> # Data Table Tutorial

Reviewer: If this isn't for the `rx.data_table` component, maybe the heading should be "Data Editor Tutorial"?

Author: yes this is something I discussed with nikhil, it is just that more commonly people refer to this component as a data table. Perhaps we could call it the data table (editable).

```python eval
rx.alert(
        rx.alert_icon(),
        rx.alert_title(f"There is another ",
        doclink("datatable component", href="https://reflex.dev/docs/library/datadisplay/datatable/"),

>         ", but this is only used for displaying data and there is no possible user interaction while using this component"),

Reviewer: ```suggestion
        ", which is only used for displaying data and does not support user interactivity or editing"),
```

        status="info",
    )
```

```python eval
rx.box(height="2em")
```

> We need to start by defining our `columns` that will then take in our data. These columns should be a list of dict, where each dict describes the associated columns. 

Reviewer: ```suggestion
We need to start by defining our columns that describe the shape of our data. The column var should be typed as a `list` of `dict` (`list[dict]`), where each item describes the attributes of a single column in the table. 
```

> Each dict has these variables below:

Reviewer: ```suggestion
Each column dict recognizes the keys below:
```

> 1. title: The text to display in the header of the column
> 2. id: An id for the column, if not defined, will default to a lower case of title
> 3. width: The width of the column
> 4. type: The type of the columns, default to "str"

Reviewer: Use code marks on the key names

```suggestion
1. `title`: The text to display in the header of the column
2. `id`: An id for the column, if not defined, will default to a lower case of title
3. `width`: The width of the column (in pixels?)
4. `type`: The type of the columns, default to "str"
```

For `type`, how can i find out what types are available?

Reviewer: are there more recognized keys for a column dict as well? where are those defined?

Author: as far as I can tell these are the only 4, but going through the reflex docs and the reflex code these are not actually defined anywhere that I can see so it is hard to say if there are more, we will need to chat to Thomas and ask

> Below we define a `DataTableState` which has Harry Potter data. Here an example of a dict in the cols var is: {
>             "title": "Name",
>             "type": "str",
>             "width": 300,
>         }.

Reviewer: ```suggestion
Below we define `DataTableState` with columns definitions in the `cols` var, and data about Harry Potter characters in the `data` var..
```

            "width": 250,
        },
        {
            "title": "Patronus",
            "type": "str",
        },
        {
            "title": "Blood status",
            "type": "str",
            "width": 200,
        },
    ]
    data = [
        ["1", "Harry James Potter", "31 July 1980", True, "Gryffindor", "11'  Holly  phoenix feather", "Stag", "Half-blood"],
        ["2", "Ronald Bilius Weasley", "1 March 1980", True,"Gryffindor", "12' Ash unicorn tail hair", "Jack Russell terrier", "Pure-blood"],
        ["3", "Hermione Jean Granger", "19 September, 1979", True, "Gryffindor", "10¾'  vine wood dragon heartstring", "Otter", "Muggle-born"],	
        ["4", "Albus Percival Wulfric Brian Dumbledore", "Late August 1881", True, "Gryffindor", "15' Elder Thestral tail hair core", "Phoenix", "Half-blood"],	
        ["5", "Rubeus Hagrid", "6 December 1928", False, "Gryffindor", "16'  Oak unknown core", "None", "Part-Human (Half-giant)"], 
        ["6", "Fred Weasley", "1 April, 1978", True, "Gryffindor", "Unknown", "Unknown", "Pure-blood"], 
        ["7", "George Weasley", "1 April, 1978", True, "Gryffindor", "Unknown", "Unknown", "Pure-blood"],
    ]
```

> We then define our simple table using `rx.data_editor()`, where we pass in the columns and data as variables.

Reviewer: ```suggestion
We then define a basic table by passing the previously defined state vars as props `columns` and `data` to the `rx.data_editor()` component, 
```

```python demo
rx.data_editor(
    columns=DataTableState.cols,
    data=DataTableState.data,
)
```

> Now we have our datatable, but there is no way to interact with it. On the next page we will explore how to add interactivity to our datatable.

Reviewer: ```suggestion
This is enough to display the data, but there is no way to interact with it. On the next page we will explore how to add interactivity to our datatable.
```
